### PR TITLE
Add extra documentation to Enum to point to EnumName.

### PR DESCRIPTION
**Pull request by Hixie**
I keep forgetting you can use `.name` because the only sign that it exists in the `Enum` documentation is the in-passing mention that `Enum` has an extension, without mentioning why. This add an explicit callout to the `Enum` class definition.

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR, as far as I can tell.

### CODE REVIEW

File: sdk/lib/core/enum.dart [hunk 1]
```diff
@@ -10,6 +10,9 @@ part of dart.core;
 /// introduced using an `enum` declaration.
 /// Non-platform classes cannot extend or mix in this class.
 /// Concrete classes cannot implement the interface.
+///
+/// The string name of a value is available via the `name` property
```
Comment (lrhn) [inline sdk/lib/core/enum.dart:14]: I'd go with 
```dart
/// The intentifier used to name an `enum` value is available as a `String`, 
/// via the [`name`](EnumName.name) extension property on the `enum` value.
```
(If that's thow you link to other members without their full name in DartDoc. Where is the DartDoc syntax specification?)

Comment (Hixie) [reply sdk/lib/core/enum.dart:14]: I'm not sure if that works, but I'm happy to change it to that if you would prefer. Let me know. :-)

**Comment by mraleph** [comment]
@Hixie would you like to follow up on the comments or should we close the PR?

**Comment by Hixie** [comment]
I was awaiting clarification about whether to apply @lrhn's suggestion or not. Is that the preferred syntax? It wasn't clear to me if that was what was being suggested or not.

**Comment by Hixie** [comment]
I applied the change but without the part that's risky (which has the side-effect of being more explicit, which is probably good anyway).

**Comment by mraleph** [comment]
@lrhn friendly ping

**Pull request closed without merging**